cpp/support/container.h

```diff
@@ -0,0 +1,115 @@
+#ifndef XGRAMMAR_SUPPORT_CONTAINER_H_
+#define XGRAMMAR_SUPPORT_CONTAINER_H_
+#include <vector>
+
+namespace xgrammar {
+
+template <typename Value>
+class list {
```

Comment:
Could you separate logic of NodePool (allocate and free) and List (random insert)